duktape/src/main/jni/duktape-jni.c

```diff
@@ -20,11 +20,18 @@ Java_com_squareup_duktape_Duktape_evaluate__JLjava_lang_String_2(JNIEnv *env,
   void* ctx = (void*) context;
   const char *s = (*env)->GetStringUTFChars(env, s_, 0);
 
-  duk_eval_string(ctx, s);
-  jstring result = (*env)->NewStringUTF(env, duk_get_string(ctx, -1));
-  duk_pop(ctx);
+  jstring result = NULL;
 
-  (*env)->ReleaseStringUTFChars(env, s_, s);
+  if (duk_peval_string(ctx, s) != 0) {
+      jclass Exception = (*env)->FindClass(env, "com/squareup/duktape/DuktapeException");
+      (*env)->ThrowNew(env, Exception, duk_safe_to_string(ctx, -1));
+      goto cleanup;
+  }
+
+  result = (*env)->NewStringUTF(env, duk_get_string(ctx, -1));
```

Comment:
Fixed to not goto. I was using it as a moral equivalent of `finally`. I don't think we want `safe` here; that's only necessary if there's an error.